portablemc/optifine.py

```diff
@@ -0,0 +1,659 @@
+    def check_of_install(self, version: VersionHandle) -> bool:
+        """
+        Check if all needed ressources are properly installed to allow standard installation.
+        """
+        if not version.read_metadata_file():
+            return False
+        if "inheritsFrom" in version.metadata and version.metadata["inheritsFrom"] == self.mcver():
+            if "libraries" in version.metadata:
+                launchwrapperseemscorrect = False # This variable is used to check if the launchwrapper is correct
+                oflibseemscorrect = False
+                for lib in version.metadata["libraries"]:
+                    libpath=LibrarySpecifier.from_str(lib["name"])
+                    if not (self.context.libraries_dir / libpath.file_path()).exists() and not "url" in lib.keys():
+                        return False
+                    else:
+                        if "sha1" in lib.keys(): # A functionality that check if libs remains the same
+                            with open(self.context.libraries_dir / libpath.file_path(), "rb") as f:
+                                data = f.read()
+                            if not sha1(data).hexdigest()==lib["sha1"]:
+                                return False
+                    if lib["name"].startswith("optifine:launchwrapper-of:") or "net.minecraft:launchwrapper:1.12" == lib["name"]:
+                        launchwrapperseemscorrect = True
+                    if lib["name"]==f"optifine:OptiFine:{self.mcver()}_{self.loader()}":
+                        oflibseemscorrect = True
+
+
+                if ("minecraftArguments" in version.metadata and
+                    "--tweakClass optifine.OptiFineTweaker" not in version.metadata["minecraftArguments"]):
+                    return False
+                elif ("arguments" in version.metadata and
+                    "game" in version.metadata["arguments"] and
+                    "--tweakClass" not in version.metadata["arguments"]["game"] and
+                    "optifine.OptiFineTweaker" not in version.metadata["arguments"]["game"]):
+                    return False
+                elif ("minecraftArguments" in version.metadata and
+                    "--tweakClass optifine.OptiFineTweaker" in version.metadata["minecraftArguments"]):
+                    return launchwrapperseemscorrect and oflibseemscorrect
+                elif ("arguments" in version.metadata and
+                    "game" in version.metadata["arguments"] and
+                    "--tweakClass" in version.metadata["arguments"]["game"] and
+                    "optifine.OptiFineTweaker" in version.metadata["arguments"]["game"]):
+                    return launchwrapperseemscorrect and oflibseemscorrect
+                else:
+                    return False
+        return False
+    def _finalize_optifine_internal(self, watcher: Watcher) -> None:
+        version = VersionHandle(self.version, self.context.versions_dir / self.version)
```

Comment:
Don't need to re-construct a version handle, simply use the hierarchy `self._hierarchy[0]` contains the top-level version handle.